p2p/transport/websocket/listener.go

```diff
@@ -54,11 +56,20 @@ func newListener(a ma.Multiaddr, tlsConf *tls.Config) (*listener, error) {
 	if err != nil {
 		return nil, err
 	}
-	nl, err := net.Listen(lnet, lnaddr)
-	if err != nil {
-		return nil, err
+	if reuseportAvailable {
+		nl, err = reuseport.Listen(lnet, lnaddr)
+		if err != nil {
```

Comment:
We want to use the transport's `reuseTransport` field. Otherwise this will never reuse the port we're listening on when dialing. I guess you can pass them in to `newListener`. 

Also add a test that we are reusing port when dialing.